### PR TITLE
Update error returned for various HttpStatus codes

### PR DESCRIPTION
**Pull request by pallavijustice**
### Change description ###

Update error message returned for certain HttpStatus codes to ensure that Liberata contact information in not unnecessarily added to the message

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-4086

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/payment/PaymentService.java [hunk 1]
```diff
@@ -326,4 +330,12 @@ public Double getServiceCost(String service, String event, String keyword) {
 
         return feeResponse.getAmount();
     }
+
+    private boolean isGenericErrorRequiredForHttpStatus(HttpStatus httpStatus) {
+        boolean required = switch (httpStatus) {
+            case INTERNAL_SERVER_ERROR, GATEWAY_TIMEOUT -> true;
+            default -> false;
+        };
+        return required;
```
Comment (adamg-hmcts) [inline src/main/java/uk/gov/hmcts/divorce/payment/PaymentService.java:339]: How about using ||? There are only 2 conditions:
```suggestion
        return httpStatus == INTERNAL_SERVER_ERROR || httpStatus == GATEWAY_TIMEOUT;
```

Comment (pallavijustice) [reply src/main/java/uk/gov/hmcts/divorce/payment/PaymentService.java:339]: Had started with this but was wondering if in future we will need more custom error messages based on HttpStatus. If that could be the case, I could have this function return the error message itself based on value of httpStatus?
The ticket however mentions only 2 codes (500, 504)

**Review by adamg-hmcts — DISMISSED** [review]

**Review by DelythJustice — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/divorce/payment/PaymentService.java [hunk 2]
```diff
@@ -186,6 +189,10 @@ private PbaResponse getPbaErrorResponse(String pbaNumber, FeignException excepti
             return new PbaResponse(httpStatus, String.format(NOT_FOUND.value(), pbaNumber), null);
         }
 
+        if (isGenericErrorRequiredForHttpStatus(httpStatus)) {
```
Comment (pallavijustice) [inline src/main/java/uk/gov/hmcts/divorce/payment/PaymentService.java:192]: Will triggered a scenario where the internal feign client for fee-and-pay isn't reachable. If that scenario is possible as well (which is likely), we will need to bring the check for the httpStatus(500, 504) codes right after the FeignException is caught to return the generic error before other conditions are checked for other possible error conditions.

File: src/main/java/uk/gov/hmcts/divorce/payment/PaymentService.java [hunk 3]
```diff
@@ -186,6 +189,10 @@ private PbaResponse getPbaErrorResponse(String pbaNumber, FeignException excepti
...
+            return new PbaResponse(httpStatus, ERROR_GENERIC, pbaNumber);
+        }
+
```
Comment (pallavijustice) [inline src/main/java/uk/gov/hmcts/divorce/payment/PaymentService.java:195]: Will triggered a scenario where the internal feign client for fee-and-pay isn't reachable. If that scenario is possible as well (which is likely), we will need to bring the check for the httpStatus(500, 504) codes right after the FeignException is caught to return the generic error before other conditions are checked for other possible error conditions.

**Review by adamg-hmcts — APPROVED** [review]

**Review by DelythJustice — APPROVED** [review]

**Pull request merged**